Solutions/CohesitySecurity/Tests/az.test.py

```diff
@@ -83,54 +103,31 @@ def test_cohesity_restore_from_last_snapshot(self):
         protection_group_id = alert.get_protection_group_id()
         cluster_id = alert.get_cluster_id()
 
-        query_range_usecs = 3 * 60 * MICROSECS_IN_SEC
-
-        # Calculate the current time and subtract 3 minutes to get a start time
-        # for the recovery.Only recoveries that start within the last 3 minutes
-        # will be retrieved.
-        current_time_usecs = int(time.time() * MICROSECS_IN_SEC)
-        start_time_usecs = current_time_usecs - query_range_usecs
-
-        recoveries = get_recoveries(cluster_id, self.api_key, start_time_usecs)
-
-        assert (
-            recoveries is not None
-            and recoveries.get("recoveries")
-            and not any(
-                # Check if there are any objects with the same protection group
-                # ID as the one obtained from the alert
-                any(
-                    obj["protectionGroupId"] == str(protection_group_id)
-                    for obj in recovery["vmwareParams"]["objects"]
-                )
-                for recovery in recoveries["recoveries"]
-            )
-        )
-
         returncode = run_playbook(
             subscription_id,
             incident_id,
             self.resource_group,
             self.workspace_name,
             playbook_name,
         )
-        self.assertEqual(returncode, 0)
 
         time.sleep(30)  # Sleep for 30 seconds
 
-        recoveries = get_recoveries(cluster_id, self.api_key, start_time_usecs)
-
-        # Check that the data is not null or empty
+        playbook_run = get_latest_playbook_run(
```

Comment:
Can you please call this inside run_playbook() and return the response directly? Every playbook just needs the response and nothing else anyway.

Ideally, you can remove the sleep and poll on the status of the playbook run till it succeeds or fails.